### DISCUSSION TITLE
✔️ Add summon with heart

### DISCUSSION THREAD

**Discussion by kevingrillet** [zebscripts/AFK-Daily#34] [Ideas]
I don't know if it's possible, but, I think this path should work:

> The Noble Tavern > Roll right until heart if found in right corner > Summon 1

**Comment by Zebiano** [comment]
Just making sure I understood the concept: You'd like the script to make a companion points summon whenever possible?

**Comment by kevingrillet** [comment]
Yes, sorry, my English is a bit rusty :p
Summon 1 hero with Companion Points is a free 20 Activity Points, if another option is not available during leveling it could be useful.

**Reply by Zebiano** [reply]
No problem! You are indeed right, I could use companion points to summon one hero in order to get those extra 20 activity points on a daily basis. Thank you for the suggestion, I will keep this in mind! 1f5ceb18aec3ebb17f05e3b5f6b73b7b0faa0611

**Reply by kevingrillet** [reply]
I will do a pull request tomorrow, but I think it works:
```shell
# Let's do a "free" summon
function nobleTavern() {
    input tap 280 1370 # The Noble Tavern
    sleep 3

    input tap 600 1820 # The noble tavern again
    sleep 1

    getColor 897 852
    until [ "$RGB" == "ecd383" ]; do # Looking for heart
        #echo "RGB: $RGB"   # for testing
        input tap 870 1630 # Next pannel
        sleep 1

        getColor 897 852
    done

    input tap 350 1450 # Summon
    sleep 3
    input tap 540 908  # Click on the card
    sleep 2
    input tap 70 1810  # close
    sleep 1
    input tap 550 1700 # if auto dez...
    sleep 1

    input tap 70 1810

    wait
    verifyRGB 20 1775 d49a61 "Companion points summon." "Failed to summon."
}
```

**Reply by Zebiano** [reply]
It does look promising indeed! I'll be waiting for the PR for testing. Thank you.

**Reply by kevingrillet** [reply]
Did try this morning and it's working properly \o/
